library/libraries.js

```diff
@@ -1919,5 +1919,30 @@ var d41d8cd98f00b204e9800998ecf8427e_LibraryDetectorTests = {
         }
         return false;
       }
+    },
+    'WP Rocket':{
+        id:'wp-rocket',
+        icon:'wp-rocket',
+        url: 'https://wp-rocket.me/',
+        npm: null,
+        test: async function (win) {
+            const wpRocketLazyLoad = typeof RocketLazyLoadScripts !== 'undefined'|| typeof RocketPreloadLinksConfig !== 'undefined' ||typeof rocket_lazy !== 'undefined';
+            const wpRocketRUCSS = !!document.querySelector('style#wpr-usedcss');
+            const wpRocketComment = document.lastChild.nodeType === Node.COMMENT_NODE && document.lastChild.textContent.includes('WP Rocket');
+
+            if ( wpRocketRUCSS || wpRocketLazyLoad || wpRocketComment ) {
+                return { version: UNKNOWN_VERSION };
+            }
+            let url = (new URL(window.location));
+            const domain = url.hostname;
+            const protocol = url.protocol;
+            // This is the path to the wp-rocket admin css.
+            const response = await fetch(`${protocol}//${domain}/wp-content/plugins/wp-rocket/assets/css/wpr-admin.min.css`);
```

Comment:
Will this ever be true if the above fails? As it is, this would be an extra fetch for _every_ page missing wp rocket, which seems like it would be undesirable (only workbox does this out of all the libs here)–so if it's not needed, best to remove.

@housseindjirdeh wdyt?

Reply:
also, if we keep this: I think we can just do `'/wp-content/plugins/wp-rocket/assets/css/wpr-admin.min.css'`–no need to repeat the protocol and domain.

Reply:
Yeah I would steer clear away from making a fetch call within the detection logic. @mostafa-hisham Are there any other signals that can be used instead (maybe reading the DOM to see if there's a `wp-rocket/assets/css/wpr-admin.min.css` stylesheet)?

Reply:
So it looks like this is just an additional check after `wpRocketRUCSS || wpRocketLazyLoad || wpRocketComment`. Could we remove this and just rely on the earlier checks here or is that not sufficient?

Reply:
^ @mostafa-hisham